var/spack/repos/builtin/packages/abacus/package.py

```diff
@@ -6,40 +6,214 @@
+        description="Enable ABACUS's builtin benchmark tests",
+        when="+tests",
+    )
+    variant("rocm", default=False, description="(Experimental)Enable rocm support")
+    # TODO: Add support for
+    # LibRI(https://github.com/abacusmodeling/LibRI),
+    # LibComm(https://github.com/abacusmodeling/LibComm),
+    # Libnpy(https://github.com/llohse/libnpy/),
+    # DeePKS(https://github.com/deepmodeling/deepks-kit),
+    # DeePMD(https://github.com/deepmodeling/deepmd-kit),
+    # LibPAW-interface(https://github.com/wenfei-li/libpaw_interface),
+    # At 2024-1-30, none of above have a spack package.
+
+    depends_on("fftw-api@3")
+    depends_on("blas")
+    depends_on("lapack")
+
+    with when("+mpi"):
+        depends_on("mpi", type=("build", "link", "run"))
+    with when("+libxc"):
+        depends_on("libxc")
+    with when("+lcao"):
+        depends_on("cereal")
+        depends_on("scalapack")
+    with when("+elpa"):
+        depends_on("elpa")
+    with when("+tests"):
+        depends_on("googletest")
+    with when("+benchmarks"):
+        depends_on("benchmark")
+    with when("+openmp"):
+        depends_on("fftw+openmp", when="^[virtuals=fftw-api] fftw")
+        depends_on("elpa+openmp", when="+elpa")
+        depends_on(
+            "openblas threads=openmp",
+            when="^[virtuals=blas] openblas" or "^[virtuals=lapack] openblas",
+        )
+    with when("~openmp"):
+        depends_on("elpa~openmp", when="+elpa")
+
+    requires("%clang" or "%rocmcc", when="+rocm", msg="build with rocm requires rocm compiler")
+
+    conflicts(
+        "^blis",
+        when="@:3.5.4",
+        msg="abacus spack package supports openblas/mkl as blas/lapack provider",
+    )
+    conflicts(
+        "^libflame",
+        when="@:3.5.4",
+        msg="abacus spack package supports openblas/mkl as blas/lapack provider",
+    )
+    conflicts(
+        "^amdblis",
+        when="@:3.5.4",
+        msg="abacus spack package supports openblas/mkl as blas/lapack provider",
+    )
+    conflicts(
+        "^amdlibflame",
+        when="@:3.5.4",
+        msg="abacus spack package supports openblas/mkl as blas/lapack provider",
+    )
+    # netlab-lapack+external-blas do NOT contain libblas.so and not detectable
+    # for abacus CMake script.
+    conflicts(
+        "^[virtuals=lapack] netlib-lapack~external-blas",
+        when="@:3.5.4",
+        msg="abacus spack package supports openblas/mkl as blas/lapack provider",
+    )
+    conflicts(
+        "^netlib-xblas",
+        when="@:3.5.4",
+        msg="abacus spack package supports openblas/mkl as blas/lapack provider",
+    )
 
-    depends_on("elpa+openmp", when="+openmp")
-    depends_on("elpa~openmp", when="~openmp")
-    depends_on("cereal")
-    depends_on("libxc")
-    depends_on("fftw")
-    # MPI is a necessary dependency
-    depends_on("mpi", type=("build", "link", "run"))
-    depends_on("mkl")
 
+class MakefileBuilder(makefile.MakefileBuilder):
```

Comment:
The signature of the method below needs to be adjusted:
```
def edit(self, pkg, spec, prefix):
    ...
```
etc.

Reply:
resolved by 484229d.